PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp

```diff
@@ -3649,6 +3665,20 @@ void FindReplaceDlg::setStatusbarMessage(const generic_string & msg, FindStatus
 	}
 }
 
+void FindReplaceDlg::setStatusbarMessageWithRegExprErr(ScintillaEditView* pEditView)
+{
+	if (!pEditView)
+		return;
+
+	char msg[511] = "";

```

Comment:
Just curious:
Is there a difference between `char msg[511] = "";` and `char msg[511]{};`?

Reply:
@Yaron10 
I do believe that, in `char msg[511] = "";` we declare a char array of 511 bytes and make the msg[0] = '\0';
whereas is `char msg[511]{};`, it fullfills 511 bytes with `\0`. 

Please let me know (anyone) if my understanding is wrong.
Therefore it's better to use `char msg[511]{};`.

I will do  another commit soon to fix it.

Reply:
@donho,

Thank you for the explanation.

Just some thoughts.

If your theory is correct - is there a reason to initialize all 511 bytes?
I understand it might be safer, but there must be some overhead to that robust initialization.
(Relevant to many other cases across the project).

In `POINT pt{}` (or `RECT rc{}`), there's no difference between `pt.x` and `pt.y`. - If it's better to initialize, both variables should be initialized.
But here we can check `msg[0]`. And I _suppose_ there's no warning if `char msg[511] = "";` is used.

> Please let me know (anyone) if my understanding is wrong.

@ozone10, @rdipardo.

Reply:
@Yaron10 
I could be wrong, but IMO, there's no overhead with modern C++ compiler initialization of `type var[512] {}` - I assume that compiler generate an array with all zero by default to you.

Reply:
@donho,

> but IMO, there's no overhead with modern C++ compiler initialization of type var[512] {}

This is above my level.
I just thought there must be _some overhead_ at runtime.
But you're the pro, and I trust your judgment. :)

> I assume that compiler generate an array with all zero by default to you.

I use MS VS.

Reply:
@Yaron10 
> And I _suppose_ there's no warning if `char msg[511] = "";` is used.
> 
> > Please let me know (anyone) if my understanding is wrong.
> 
Since you asked, `char msg[511] = ""` is not only a warning but a hard failure.

`""` is a __*pointer*__ to an array of `char`, but `msg[511]` is an actual `char`:

    error C2440: '=': cannot convert from 'const char [1]' to 'char'

Reply:
@rdipardo,

Thank you for replying.

> error C2440: '=': cannot convert from 'const char [1]' to 'char'

So how did the first commit not fail? :)